### DISCUSSION TITLE
[DONE] Integration of GPS data

### DISCUSSION THREAD

**Discussion by rwbru** [KL-Psychological-Methodology/ESMira#6] [Ideas]
Hello everyone,

we've been exploring the platform and are impressed by its possibilities. For a project, we're currently looking to integrate location data with certain questionnaires.

At least on iOS, the compass item already seems to be connected to Apple's Core Location framework, but only orientation data is pulled. Would it be possible to extend the access and also pull the devices geographic location this way? In our case, GPS data would only be needed every time people answer a certain survey type, so the request could possibly be handled the same way the compass item is currently implemented. 

Anything on the roadmap regarding this feature? Thank you for the outstanding work that has already been done on this project!

**Comment by JodliDev** [comment]
Hello.
Yes that has been on our roadmap for a while now. The main reason why we have not implemented it yet, is that GPS positioning has a lot of ramifications for anonymity (i.e it would be quite easy to guess someones home address). One idea was to use something like H3 (https://www.uber.com/en-AT/blog/h3/) to anonimize data a bit - but that needs to be implemented as well.
There is also the question if people should be able to edit their entry, which would need integration of some Android / Apple services.

In general it should not be too hard to implement. There are just a few open questions that could have the potential of making things more tricky ;)

So it is on our ToDo-list and there have already been meetings about it, but it is not on our short term todo list (and recently we had some internal changes. So for the next few months, our fokus is less on new features).

When would you need such a feature?

**Reply by rwbru** [reply]
Thank you for the detailed response and especially for considering a privacy-by-design integration. That would definitely be a major asset to ESMira. Our project aims to start with geolocated surveys around April/May this year, but given what you've described, this might not work out. We're nevertheless happy to see this feature on the roadmap and are looking forward to how it evolves!

**Reply by JodliDev** [reply]
Just that you are aware: Last week we went through aour short term priorities and since the GPS item would be usefull for us as well, we decided to push it and are trying to implement it soon. There is a chance it might be ready in time for your study. But please dont count on it yet (or at least have a viable plan B) because we might run into issues and also can't really say for sure how long the implementation will take us :)

**Reply by rwbru** [reply]
Great news!

**Comment by SelinaDev** [comment]
Hi!

Just wanted to inform you that we are currently actively implementing a location item. So far things are looking good, the item seems to work on Android, and today I finished the iOS implementation, which also seems promising. We still have to do some testing before we can officialy release an update with these items. However, unless that reveals any major issues, I am relatively confident that the location item will be ready soon.
